### ISSUE TITLE
PKCS12_parse just contain a password input

### ISSUE THREAD

**Issue by chensqecc** [openssl/openssl#7936]
hi ,i want ask a question about PFX CERT.
As we know PFX CERT can generate some pem/asn cert and keys, while here need input two password: one is enc password and another is mac password. while when load PFX cert  ,so it should pass two password too.

Now as we confirmed, openssl1.1.1 just have PKCS12_parse api can load PFX cert. while it just support one password, so it demands mac and enc password same . but actually these two passwords are not same sometimes.

so please give some suggestions that how i can load a  PFX cert who's passwords are not same?

**Comment by levitte** [comment]
With the current API, I'm afraid you will have to do the unpacking yourself, in smaller pieces.  The easiest might be to have a look at what `crypto/pkcs12/p12_kiss.c` does...

**Comment by chensqecc** [comment]
yes, now  i am trying to do this thing. i try to add a new password to PKCS12_parse ,such as :+1: int PKCS12_parse_ex(PKCS12 *p12, const char *macpass, const char *encpass,EVP_PKEY **pkey, X509 **cert,
                 STACK_OF(X509) **ca)
And the problem is parse_pk12/parse_bags/parse_bag is inner fucntion , i have to implement them again.

i want ask whether openssl1.1.1 mainline have any plan to support this feature later?

**Comment by nhorman** [comment]
It seems like we don't plan to support this in the near term future @chensqecc .  Are you interested in writing up a PR for this?  If so, I'm happy to assign this to you as a community issue

**Comment by t8m** [comment]
I do not think this use-case is important enough to provide a high-level API for it. It can still be handled via lower level API calls.
